pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1081,7 +1081,13 @@ func PodFitsHostPorts(pod *v1.Pod, meta PredicateMetadata, nodeInfo *schedulerno
 
 // search two arrays and return true if they have at least one common element; return false otherwise
 func haveOverlap(a1, a2 []string) bool {
-	m := map[string]bool{}
+        // use the longer array to fill the map so that expected number of comparisons is lower
```

Comment:
I know this is theoretical, but don't we want the shorter array to populate the map? In the two for-loops below, `m[val]` does a hash & lookup in the map. In the worst case, we have to do this `len(a1)+len(a2)` times. But if the map is smaller then we allocate less memory.

Anyway, if you can show with benchmarks that a change like this improves scheduler performance, then we're happy to reopen & review this.